syntax = "proto3";

package syft.grid.messages;

import "proto/core/common/common_object.proto";
import "proto/core/io/address.proto";
import "proto/lib/python/dict.proto";


// CREATE User
message CreateUserMessage {
  syft.core.common.UID msg_id = 1;
  syft.core.io.Address address = 2;
  string email = 3;
  string password = 4;
  string role = 5;
  string name = 6;
  string website = 9;
  string institution = 10;
  bytes daa_pdf = 11;
  syft.core.io.Address reply_to = 12;
  double budget = 13;
}

// GET User
message GetUserMessage {
  syft.core.common.UID msg_id = 1;
  syft.core.io.Address address = 2;
  int32 user_id = 3;
  syft.core.io.Address reply_to = 4;
}

message ProcessUserCandidateMessage {
  syft.core.common.UID msg_id = 1;
  int32 candidate_id = 2;
  string status = 3;
  syft.core.io.Address address = 4;
  syft.core.io.Address reply_to = 5;
}

message GetUserResponse {
  syft.core.common.UID msg_id = 1;
  syft.lib.python.Dict content = 2;
  syft.core.io.Address address = 3;
}

// GET USER Requests
message GetCandidatesMessage {
  syft.core.common.UID msg_id = 1;
  syft.core.io.Address address = 2;
  syft.core.io.Address reply_to = 4;
}

message GetCandidatesResponse {
  syft.core.common.UID msg_id = 1;
  repeated syft.lib.python.Dict content = 2;
  syft.core.io.Address address = 3;
}

// GET ALL Users
message GetUsersMessage {
  syft.core.common.UID msg_id = 1;
  syft.core.io.Address address = 2;
  syft.core.io.Address reply_to = 4;
}

message GetUsersResponse {
  syft.core.common.UID msg_id = 1;
  repeated syft.lib.python.Dict content = 2;
  syft.core.io.Address address = 3;
}

// Delete User
message DeleteUserMessage {
  syft.core.common.UID msg_id = 1;
  syft.core.io.Address address = 2;
  int32 user_id = 3;
  syft.core.io.Address reply_to = 4;
}

// UPDATE User
message UpdateUserMessage {
  syft.core.common.UID msg_id = 1;
  syft.core.io.Address address = 2;
  int32 user_id = 3;
  string email = 4;
  string password = 5;
  string role = 6;
  string groups = 7;
  string name = 8;
  double budget = 9;
  string institution = 10;
  string website = 11;
  string new_password = 12;
  syft.core.io.Address reply_to = 13;
}

// Search Users
message SearchUsersMessage {
  syft.core.common.UID msg_id = 1;
  syft.core.io.Address address = 2;
  string email = 3;
  string role = 4;
  string groups = 5;
  string name = 6;
  syft.core.io.Address reply_to = 7;
}

message SearchUsersResponse {
  syft.core.common.UID msg_id = 1;
  repeated syft.lib.python.Dict content = 2;
  syft.core.io.Address address = 3;
}
